### ISSUE TITLE
When someone calls detach() on a __torch_dispatch__ object, detach() gets called twice

### ISSUE THREAD

**Issue by zou3519** [pytorch/pytorch#71725]
### 🐛 Describe the bug

See https://gist.github.com/zou3519/6a673f7b624ca2365361e45ce7843439 .
```
with capture_logs() as logs:
    x = LoggingTensor(torch.tensor([3.0], requires_grad=True))
    log_input("x", x)
    x.detach()
print(logs)
```
prints:
```
["$0 = input('x')", '$1 = torch._ops.aten.detach($0)', '$2 = torch._ops.aten.detach($1)']
```
on PyTorch 1.10 and PyTorch master

### Versions

1.10 and master

cc @albanD @zou3519

**Comment by zou3519** [comment]
cc @albanD @soulitzer since we were discussing this earlier today.

I figured out why detach() prints twice: it's because shallow_copy_and_detach is being called twice!
- shallow_copy_and_detach gets called once [here ](https://github.com/pytorch/pytorch/blob/f75e92a936f32303b3fa570a6fab144aa0acec44/torch/csrc/autograd/VariableTypeManual.cpp#L320)
- Inside [as_view](https://github.com/pytorch/pytorch/blob/f75e92a936f32303b3fa570a6fab144aa0acec44/torch/csrc/autograd/VariableTypeManual.cpp#L325), it gets called again in [make_variable_non_differentiable_view](https://github.com/pytorch/pytorch/blob/master/torch/csrc/autograd/variable.h#L668-L670)

The first shallow_copy_and_detach is probably unnecessary.

**Comment by albanD** [comment]
Feels like we have some overdue work with removal of make_variable...
